### PR TITLE
Montana/experimental hsa

### PR DESCRIPTION
**Pull request by montycheese**

### CODE REVIEW

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 1]
```diff
@@ -35,10 +52,17 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
       preference,
     });
 
+    if (this.preference.enableAutomaticSubAccountCreation) {
```
Comment (cb-jake) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:55]: @montycheese which 3rd party lib was this aiming to fix? I think we can move this up into the CoinbaseWalletSDK class vs the provider but want to make sure I test that edge case

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 2]
```diff
@@ -47,21 +71,89 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
       if (!this.signer) {
         switch (args.method) {
           case 'eth_requestAccounts': {
-            const signerType = await this.requestSignerSelection(args);
+            // this causes a popup which we dont want.
+            //const signerType = await this.requestSignerSelection(args);
+            const signerType = 'scw';
```
Comment (cb-jake) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:76]: we still need the popup so users can select whether or not they want to use a scw or walletlink

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts (hunk 1 above)
Comment (montycheese) [reply packages/wallet-sdk/src/CoinbaseWalletProvider.ts:55]: it was wagmi. 

it was just a way to set the default toSubAcccountSigner impl if one wasn't being provided, since wagmi doesn't currently see all of our sdk params, just the preferences map.

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts (hunk 2 above)
Comment (montycheese) [reply packages/wallet-sdk/src/CoinbaseWalletProvider.ts:76]: I may have just removed this in the moment because it was annoying
